chatt-app/ChatClient.java

```diff
@@ -0,0 +1,155 @@
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.event.KeyEvent;
+import java.awt.event.KeyListener;
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+import java.io.*;
+import java.net.*;
+
+import javax.swing.*;
+
+
+public class ChatClient extends JFrame implements Runnable {
+	
+	Socket socket;
+	JTextArea ta;
+	JButton send, logout;
+	JTextField tf;
+	
+	Thread thread;
+	
+	DataInputStream din;
+	DataOutputStream dout;
+	
+	String LoginName;
+	
+	ChatClient(String login) throws UnknownHostException, IOException{
+		super(login);
+		LoginName = login;
+		
+		addWindowListener(new WindowAdapter(){
+			public void windowClosing(WindowEvent e){
+				try {
+					dout.writeUTF(LoginName + " " + "LOGOUT");
+					System.exit(1);
+				} catch (IOException e1) {
+					// TODO Auto-generated catch block
+					e1.printStackTrace();
+				}
+			}
+		});
+		
+		ta = new JTextArea(18, 50);
+		tf = new JTextField(50);
+		
+		tf.addKeyListener(new KeyListener(){
+
+			@Override
+			public void keyTyped(KeyEvent e) {
+				// TODO Auto-generated method stub
+				
+			}
+
+			@Override
+			public void keyPressed(KeyEvent e) {
+				// TODO Auto-generated method stub
+				if(e.getKeyCode() == KeyEvent.VK_ENTER){
+					try {
+						if(tf.getText().length()>0)
+							dout.writeUTF(LoginName + " " + "DATA " + tf.getText().toString());
+						tf.setText("");
+					} catch (IOException e1) {
+						// TODO Auto-generated catch block
+						e1.printStackTrace();
+					}
+				}
+			}
+
+			@Override
+			public void keyReleased(KeyEvent e) {
+				// TODO Auto-generated method stub
+				

```

Comment:
Omission (O) - Assuming this package is completed code, the keyReleased function was never completed. 

Rank 2 - Fix as part of planned debugging sessions